File: FoodieApplication/RestuarantManagementService/src/main/java/com/example/RestuarantManagementService/model/Restaurant.java
package com.example.RestuarantManagementService.model;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document
public class Restaurant {

    @Id
    private String restaurantId;
    private String restaurantName;
    private String restaurantLocation;
    private List<Dish> dishList;
    private String status = "reject";

    public Restaurant(String restaurantId, String restaurantName, String restaurantLocation, List<Dish> dishList) {
    }
}

File: FoodieApplication/frontend/FoodieApplication/src/app/favService/domain/restaurant.ts
import { Dish } from "./dish";
import { Image } from "./image";

export class Restaurant {

    restaurantId:string;
    restaurantName:string;
    restaurantLocation:string;
    image:Image[];
    dishList: Array<Dish>;

    constructor(){
        this.restaurantId='',
        this.restaurantName='';
        this.restaurantLocation='';
        this.image=[];
        this.dishList=[];
    }
}
